File: 2015/day01-2.js
import fs from 'fs'

const data = fs.readFileSync('./data/input.txt', { encoding: 'utf-8' })
const chars = data.split('')

let floor = 0
let i = 1
for (const current of chars) {
  if (current === '(') {
    floor++
  }
  else {
    floor--
  }
  if (floor === -1) {
    break
  }
  else {
    i++
  }
}


console.log(i)

File: 2015/day01-2.go
package main

import (
	"fmt"
	"os"
)

func main() {
	content, err := os.ReadFile("./data/input.txt")
	if err != nil {
		fmt.Println(err)
		return
	}

	sum := 0
	for i := 0; i < len(content); i++ {
		if content[i] == 40 {
			sum++
		} else {
			sum--
		}

		if sum == -1 {
			fmt.Println(i + 1)
			break
		}

	}
}
